adapters/tagoras/tagoras.go

```diff
@@ -0,0 +1,132 @@
+package tagoras
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+	"net/http"
+	"net/url"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the tagoras for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requests []*adapters.RequestData
+	var errors []error
+
+	requestCopy := *request
+
+	for _, imp := range request.Imp {
+		requestCopy.Imp = []openrtb2.Imp{imp}
+
+		requestJSON, err := json.Marshal(&requestCopy)
+		if err != nil {
+			errors = append(errors, fmt.Errorf("marshal bidRequest: %w", err))
```

Comment:
Can we direclty return from here as it will fail for the next impression as well?